nwbinspector/checks/time_series.py

```diff
@@ -92,3 +92,38 @@ def check_resolution(time_series: TimeSeries):
         return InspectorMessage(
             message=f"'resolution' should use -1.0 or NaN for unknown instead of {time_series.resolution}."
         )
+
+
+@register_check(importance=Importance.BEST_PRACTICE_SUGGESTION, neurodata_type=TimeSeries)
+def check_rows_not_nan(time_series: TimeSeries, nelems=200):
+    """Check that each row of a TimeSeries has at least one non-NaN piece of data."""
+    n_dims = len(time_series.data.shape)
+    if n_dims > 2:
+        yield
+
+    subframe_selection = np.unique(
+        np.round(np.linspace(start=0, stop=time_series.data.shape[0] - 1, num=nelems)).astype(int)
+    )
+    if n_dims == 1:
+        if not all(np.isnan(time_series.data[:nelems])):
+            yield
+
+        if all(np.isnan(time_series.data[subframe_selection])):
+            yield InspectorMessage(
+                message=(
+                    "This TimeSeries appears to contain NaN data at each frame. "
+                    "Consider removing this object from the file."
+                )
+            )
+    elif n_dims == 2:
+        for col in range(time_series.data.shape[1]):
```

Comment:
what if there are 1000s of columns?

Reply:
3 SpikeGLX probes would hit that amount, not to mention volumetric ophys data...

Could be faster if we did the `np.isnan` calculation vectorized over the entire extracted array of `[subselected_rows, all_cols]`...

Same case could be made for the `check_table_cols_nan` for DynamicTables (could be thousands of columns there that we iterate over), but I get that the TimeSeries data is likely to be much larger.

If you're suggesting to subselect over columns as well then here, I'd just point out that could ultimately reduce the accuracy if there are only a small number of `channels/ROIs` that had nothing in them but NaN at every time point - the likelihood of those offending columns being selected may not be very high, so I'd actually prefer limiting the `nelems` of the row data selection to examine fewer data values per row of each column...

Now, that's aside from the other idea I have for pulling data chunk-wise whenever possible, and fully utilizing each chunks data (described below).